experimental/dds/tree2/src/core/tree/anchorSet.ts

```diff
@@ -6,10 +6,11 @@
 import { assert } from "@fluidframework/common-utils";
 import { createEmitter, ISubscribable } from "../../events";
 import { brand, Brand, fail, Invariant, Opaque, ReferenceCountedBase } from "../../util";
-import { FieldKey, EmptyKey, Delta, visitDelta, DeltaVisitor } from "../tree";
 import { UpPath } from "./pathTree";
-import { Value, detachedFieldAsKey, DetachedField } from "./types";
+import { Value, detachedFieldAsKey, DetachedField, FieldKey, EmptyKey } from "./types";
 import { PathVisitor } from "./visitPath";
+import { visitDelta, DeltaVisitor } from "./visitDelta";
+import type { Delta } from ".";
```

Comment:
I thought we were avoiding imports from ".".

Reply:
```suggestion
import type * as Delta from "./delta";
```

Reply:
This is bad, but its less bad than what it did before (import lots of things including non-types from "." except "." was misspelled "../tree"): Cyclic type imports only hurt us developers and not our customers, so its an improvement.

Reply:
Now that I look at this more deeply, I guess `import * as Delta from "./delta";` might be better?

Reply:
Changed.

Reply:
@Abe27342  would be cool if the GitHub UI made your message show before I refreshed the page :), but I guess you agree with what I independently decided.

Reply:
I think we can get the best of both worlds with `import type *` rather than `import *`:

corresponding .js:
![image](https://github.com/microsoft/FluidFramework/assets/5222607/befd7ab4-f30d-4fba-b1d3-3e86b2ef6ed1)

and .d.ts:
![image](https://github.com/microsoft/FluidFramework/assets/5222607/c4093501-fc50-47a6-b98f-6c7ca6587024)